src/win/tty.c

```diff
@@ -2293,7 +2326,17 @@ void uv__tty_endgame(uv_loop_t* loop, uv_tty_t* handle) {
 
 
 int uv_tty_reset_mode(void) {
-  /* Not necessary to do anything. */
+  /**
+   * Shells on Windows do know to reset output flags after a program exits,
+   * but not necessarily input flags, so we do that for them.
+   */
+  if (
+    uv__tty_console_handle_in != INVALID_HANDLE_VALUE &&
+    uv__tty_console_in_original_mode != (DWORD)-1 &&
+    InterlockedOr(&uv__tty_console_in_need_mode_reset, 0) != 0
```

Comment:
Shouldn't this be some kind of compare-and-exchange? Otherwise multiple uv_tty_reset_mode calls result in multiple SetConsoleMode calls, right? Maybe that's harmless but it's not completely obvious looking at this code in isolation.

Reply:
Yeah, good point -- should be harmless, but also what you're describing is the behavior that I'd expect. Done in c785048904b99f42f873c861cf556aa592394d84